gammapy/catalog/fermi.py

```diff
@@ -44,6 +44,21 @@ def compute_flux_points_ul(quantity, quantity_errp):
     return 2 * quantity_errp + quantity
 
 
+def format_flux_points_table(table):
+    """Returns formatted version of a flux points table"""
+    table = table.copy()
+
+    for column in table.colnames:
+        if column in ["sqrt_TS", "e_ref", "e_min", "e_max"]:
+            table[column].format = ".1f"
+        elif column == "is_ul":
+            continue
+        else:
+            table[column].format = ".3"
```

Comment:
This will raise a ValueError if someone has e.g. an extra bool column `publish_point` or whatever in the table.
repr is used for debugging, so having repr error out with a ValueError when someone has a weird table isn't nice.
Generally I think our philosophy everywhere in Gammapy it so access some keys in meta and columns in table, and to just ignore the rest.

So even if it's a bit more work to type, I'd suggest to have an explicit list of columns to format and do nothing with the rest.

There might be a possibility to reduce a bit of code duplication and have a helper function that's used in `gammapy.catalog` as well. I didn't look in detail, but there we have this:
```
$ ack '\.format = ' gammapy/catalog
gammapy/catalog/gammacat.py
389:        table["e_ref"].format = ".3f"
392:            table[_].format = ".3e"

gammapy/catalog/fermi.py
390:        table["sqrt_TS"].format = ".1f"
391:        table["e_ref"].format = ".1f"
393:            table[_].format = ".3"
883:        table["sqrt_ts"].format = ".1f"
884:        table["e_ref"].format = ".1f"
886:            table[_].format = ".3"

gammapy/catalog/hess.py
418:            flux_points[_].format = ".3f"
421:            flux_points[_].format = ".3e"
``

Reply:
I decided to add a `FluxPoints.table_formatted` property and only applied the formatting to a certain set of columns.